functions/src/tests/index.test.ts

```diff
@@ -0,0 +1,87 @@
+  TEST_SHARE_LINK_URL,
+  registerUrl,
+  INVALID_URL_ERROR,
+  URL_NOT_FOUND_ERROR,
+} from "./utils";
+
+describe("POST /registerUrl/", () => {
+  test("it returns a 200 and creates a new share link from valid params.", async () => {
+    // the TEST_PAYLOAD params are already registered in setup.ts, so use a different url.
+    const newPayload = { ...TEST_PAYLOAD, url: "https://cdc.gov" };
+    const expectedId = createUniqueId(JSON.stringify(newPayload));
+    const response = await registerUrl(newPayload);
+    const json = await response.json();
+    expect(response.ok).toBe(true);
+    expect(json).toMatchObject({ url: `${API_BASE_URL}/go/${expectedId}` });
+  });
+
+  test("it returns the existing share link if one already exists the given params.", async () => {
+    const response = await registerUrl(TEST_PAYLOAD); // Re-register the TEST_PAYLOAD share link.
+    const json = await response.json();
+    expect(response.ok).toBe(true);
+    expect(json).toMatchObject({ url: TEST_SHARE_LINK_URL });
+  });
+
+  test.each([undefined, "ftp://not-a-valid-url"])(
+    "it returns a 400 error for a missing or invalid url.",
+    async (url) => {
+      const response = await registerUrl({ ...TEST_PAYLOAD, url: url });
+      expect(response.status).toBe(INVALID_URL_ERROR.httpCode);
+    }
+  );
+});
+
+describe("GET /go/:id", () => {
+  test("it returns a 404 for a non-existing share link.", async () => {
+    const url = `${API_BASE_URL}/go/nonexistent-id`;
+    const response = await fetch(url);
+    expect(response.status).toBe(URL_NOT_FOUND_ERROR.httpCode);
+  });
+
+  test("it returns a 200 for a valid share link.", async () => {
+    const response = await fetch(TEST_SHARE_LINK_URL);
+    expect(response.ok).toBe(true);
+  });
+});
+
+describe("GET /shareLinksByUrl", () => {
+  test("it returns expected data for a url with a share link.", async () => {
+    const url = `${API_BASE_URL}/shareLinksByUrl?url=${TEST_PAYLOAD.url}`;
+    const response = await fetch(url);
+    const json = await response.json();
+    expect(response.ok).toBe(true);
+    expect(json).toMatchObject({
+      urls: { [TEST_SHARE_LINK_URL]: TEST_PAYLOAD },
+    });
+  });
+
+  test("it returns an empty object for a url with no share links.", async () => {
+    const url = `${API_BASE_URL}/shareLinksByUrl?url=https://wwww.no-link.com`;
+    const response = await fetch(url);
+    const json = await response.json();
+    expect(response.ok).toBe(true);
+    expect(json).toMatchObject({ urls: {} });
+  });
+
+  test.each(["", "?url=ftp://not-a-valid-url"])(
+    "it returns a 400 error for a missing or invalid url.",
+    async (urlParam) => {
+      const url = `${API_BASE_URL}/shareLinksByUrl${urlParam}`;
+      const response = await fetch(url);
+      expect(response.status).toBe(INVALID_URL_ERROR.httpCode);
+    }
+  );
+});
+
+describe("GET /screenshot", () => {
+  jest.setTimeout(10000); // Bump timeout up to 10 seconds. Screenshots can take a while.
```

Comment:
The first time I ran the tests, this timed out.  The second time it succeeded.  Maybe bump it up to 15 or 20 seconds? 😬

Reply:
yikes :( thanks for the flag--I've bumped it to 15